pyartifactory/objects.py

```diff
@@ -399,31 +407,99 @@ def delete(self, name: str) -> None:
         """
         self.get(name)
         self._delete(f"api/{self._uri}/{name}")
-        logging.debug("Group %s successfully deleted", name)
+        logger.debug("Group %s successfully deleted", name)
 
 
 class ArtifactoryRepository(ArtifactoryObject):
     """Models an artifactory repository."""
 
     _uri = "repositories"
 
+    # Repositories operations
+    def get_repo(self, repo_name: str) -> AnyRepositoryResponse:
+        """
+        Finds repository in artifactory. Raises an exception if the repo doesn't exist.
+        :param repo_name: Name of the repository to retrieve
+        :return: Either a local, virtual or remote repository
+        """
+        try:
+            response = self._get(f"api/{self._uri}/{repo_name}")
+            repo: AnyRepositoryResponse = parse_obj_as(
+                Union[
+                    LocalRepositoryResponse,
+                    VirtualRepositoryResponse,
+                    RemoteRepositoryResponse,
+                ],
+                response.json(),
+            )
+            return repo
+        except requests.exceptions.HTTPError as error:
+            if error.response.status_code == 404 or error.response.status_code == 400:
+                logger.error("Repository %s does not exist", repo_name)
+                raise RepositoryNotFoundException(
+                    f" Repository {repo_name} does not exist"
+                )
+            raise ArtifactoryException from error
+
+    def create_repo(
+        self, repo: AnyRepositoryResponse
+    ) -> Union[
+        LocalRepositoryResponse, VirtualRepositoryResponse, RemoteRepositoryResponse
+    ]:
+        """
+        Creates a local, virtual or remote repository
+        :param repo: Either a local, virtual or remote repository
+        :return: LocalRepositoryResponse, VirtualRepositoryResponse or RemoteRepositoryResponse object
+        """
+        repo_name = repo.key
+        try:
+            self.get_repo(repo_name)
+            logger.error("Repository %s already exists", repo_name)
+            raise RepositoryAlreadyExistsException(
+                f"Repository {repo_name} already exists"
+            )
+        except RepositoryNotFoundException:
+            self._put(f"api/{self._uri}/{repo_name}", json=repo.dict())
+            logger.debug("Repository %s successfully created", repo_name)
+            return self.get_repo(repo_name)
+
+    def update_repo(
+        self, repo: AnyRepositoryResponse
+    ) -> Union[
+        LocalRepositoryResponse, VirtualRepositoryResponse, RemoteRepositoryResponse
+    ]:
```

Comment:
Same here, you've inverted the types
```suggestion
    def update_repo(
        self, repo: Union[LocalRepository, VirtualRepository, RemoteRepository]
    ) -> AnyRepositoryResponse:
```